File: 2021/javascript/day2/puzzle.test.js
const puzzle = require('./puzzle')
const data = require('./input')

test('partOne examples', () => {
	expect(puzzle.partOne([['forward', '5'], ['down', '5'], ['forward', '8'], ['up', '3'], ['down', '8'], ['forward', '2']])).toBe(150);
});

test('partOne', () => {
	const d = puzzle.parseData(data);
	expect(puzzle.partOne(d)).toBe(1660158);
});

test('partTwo examples', () => {
	expect(puzzle.partTwo([['forward', '5'], ['down', '5'], ['forward', '8'], ['up', '3'], ['down', '8'], ['forward', '2']])).toBe(900);
});

test('partTwo', () => {
	const d = puzzle.parseData(data);
	expect(puzzle.partTwo(d)).toBe(1604592846);
});

File: 2021/java/src/test/java/aoc/day2/PuzzleTest.java
package aoc.day2;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class PuzzleTest {
	private Puzzle puzzle;
	private static final List<Instruction> EXAMPLE_INPUT_1 = Arrays.asList(
		new Instruction("forward", 5),
		new Instruction("down", 5),
		new Instruction("forward", 8),
		new Instruction("up", 3),
		new Instruction("down", 8),
		new Instruction("forward", 2));

	@Test
	public void testPart1Example() {
		puzzle = new Puzzle(EXAMPLE_INPUT_1);
		assertEquals(150, puzzle.partOne());
	}

	@Test
	public void testPart1() {
		puzzle = new Puzzle();
		assertEquals(1660158, puzzle.partOne());
	}

	@Test
	public void testPart2Example() {
		puzzle = new Puzzle(EXAMPLE_INPUT_1);
		assertEquals(900, puzzle.partTwo());
	}

	@Test
	public void testPart2() {
		puzzle = new Puzzle();
		assertEquals(1604592846, puzzle.partTwo());
	}
}
